### PR TITLE
Cable net app/membrane ring research

### PR DESCRIPTION
**Pull request by KlausBSautter**
@philbucher @inigolcanalejo @BasselSaridar

### CODE REVIEW

**Comment by inigolcanalejo** [comment]
@KlausBSautter very nice work! Shouldn't there be some small unit tests to make sure that the element is not broken in the future?

Also just for my understanding, in this PR you include not only the membrane element but also a beam element and a utility to project the global pres-stress to the local element orientation. Is that right?

**Comment by KlausBSautter** [comment]
@inigolcanalejo the beam element is here by accident. This should be another pull request. I will change this. The unit test is needed, that's true.
We also have to discuss how to proceed with the current `PreStressMembraneElement` which does not fully work.

File: applications/StructuralMechanicsApplication/custom_elements/membrane_elment.cpp [hunk 1]
```diff
@@ -0,0 +1,1100 @@
...
+            Matrix transformation_matrix = ZeroMatrix(3);
+            InPlaneTransformationMatrix(transformation_matrix,rTransformedBaseVectors,local_prestress_axis);
+            pre_stress = prod(transformation_matrix,pre_stress);
+        }
+    }
+    rStress += pre_stress;
+}
+
+void MembraneElement::StressPk2(Vector& rStress,
+    const Matrix& rReferenceContraVariantMetric,const Matrix& rReferenceCoVariantMetric,const Matrix& rCurrentCoVariantMetric,
```
Comment (inigolcanalejo) [inline applications/StructuralMechanicsApplication/custom_elements/membrane_elment.cpp:342]: I think rReferenceContraVariantMetric is not used in this function so I guess we don't need to pass it.

**Comment by KlausBSautter** [comment]
@BasselSaridar thanks for the test. Some comments:

- please do only check 1 node
- you don't need the python file (please look at `structural_mechanics_test_factory.py` & `test_StructuralMechanicsApplication.py`)

**Comment by philbucher** [comment]
@BasselSaridar the mdpa you added is way too large for the main repo, please reduce it to 2000 lines max (preferably less)

File: applications/StructuralMechanicsApplication/tests/test_StructuralMechanicsApplication.py [hunk 2]
```diff
@@ -252,144 +252,144 @@ def AssembleTestSuites():
...
-    nightSuite.addTest(TSprismBendingPatchTests('test_execution'))
-    # Mixed displacement - volumetric strain tests
-    nightSuite.addTest(TTestCookMembrane('test_cook_membrane_2d'))
-    nightSuite.addTest(TTestCookMembrane('test_cook_membrane_incompressible_2d'))
-    # Membrane tests
-    nightSuite.addTest(TFofi4PointTentnoCableTests('test_execution'))
-    nightSuite.addTest(TFofi4PointTentCableTests('test_execution'))
-    nightSuite.addTest(TMembraneQ4PointLoadTests('test_execution'))
-    nightSuite.addTest(TMembraneQ4TrussPointLoadTests('test_execution'))
+    # smallSuite.addTests(KratosUnittest.TestLoader().loadTestsFromTestCases([TTestConstitutiveLaw]))
```
Comment (KlausBSautter) [inline applications/StructuralMechanicsApplication/tests/test_StructuralMechanicsApplication.py:255]: these are comments now

File: applications/StructuralMechanicsApplication/tests/hemisphere_membrane_test/ProjectParameters.json [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+                "model_part_name"  : "Structure",
+                "echo_level"      : 1,
+                "projection_type"        : "radial",
+                "global_direction" : [0,0,1],
+                "variable_name"    : "LOCAL_PRESTRESS_AXIS_1"
+
+            }
+        }]
+    }
+}
```
Comment (KlausBSautter) [inline applications/StructuralMechanicsApplication/tests/hemisphere_membrane_test/ProjectParameters.json]: there is no check_results process

**Pull request closed without merging**